### PR TITLE
[fix] handle empty args in chain_matmul

### LINKED ISSUE #41817: Segfault when passing an empty input to torch.chain_matmul
## 🐛 Bug

A segmentation fault occurs when passing an empty input to `torch.chain_matmul`

## To Reproduce

```python
import torch

torch.chain_matmul()
```

## Expected behavior

Seems other functions (e.g., `torch.broadcast_tensors()`) do handle an empty input without causing a segfault. I would expect the same for this function.

## Environment

PyTorch version: 1.5.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Ubuntu 18.04.4 LTS
GCC version: Could not collect
CMake version: version 3.14.0

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] torch==1.5.0
[pip] torchvision==0.6.0a0+82fd1c8
[conda] blas 1.0 mkl
[conda] cudatoolkit 10.1.243 h6bb024c_0
[conda] mkl 2020.1 217
[conda] mkl-include 2020.1 217
[conda] mkl-service 2.3.0 py37he904b0f_0
[conda] mkl_fft 1.0.15 py37ha843d7b_0
[conda] mkl_random 1.1.0 py37hd6b4f25_0
[conda] numpy 1.18.1 py37h4f9e942_0
[conda] numpy-base 1.18.1 py37hde5b4d6_1
[conda] pytorch 1.5.0 py3.7_cuda10.1.243_cudnn7.6.3_0 pytorch
[conda] torchvision 0.6.0 py37_cu101 pytorch

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes #41817

### CODE REVIEW

**Comment by mruberry** [comment]
Hey @kshitij12345! Would you add a test for this behavior to ensure it doesn't regress?

File: test/test_torch.py [hunk 1]
```diff
@@ -7237,6 +7237,9 @@ def run_test(p):
         run_test([10, 20, 30, 5])
         run_test([15, 5, 10, 20, 25])
 
+        with self.assertRaisesRegex(RuntimeError, "chain_matmul: Expected one or more matrices"):
```
Comment (kshitij12345) [inline test/test_torch.py:7240]: Have updated the existing test here. To verify the behaviour.

Comment (kshitij12345) [reply test/test_torch.py:7240]: https://github.com/pytorch/pytorch/blob/5fb52443abe463dc8c33b4e20ceb8cd98b0d77bc/test/test_torch.py#L7224-L7241

**Review by mruberry — APPROVED** [review]
Awesome! Thanks @kshitij12345!

**Pull request closed without merging**